TFNetworkLayer.py

```diff
@@ -4697,7 +4697,7 @@ def __init__(self, n_out, filter_size, padding, strides=1, dilation_rate=1,
     if input_add_feature_dim:
       # Add a feature dimension; any other static dims will be used as dynamic dims below.
       input_data = input_data.copy_add_feature_dim()
-    if self.output.is_batch_feature_major:
+    if self.output.is_batch_feature_major and TFUtil.is_gpu_available():
```

Comment:
This is wrong. The output *must* be as defined. The output should be correctly defined in `get_out_type_from_opts`.

Reply:
Yes, you are right that this is redundant code, as the output will be set correctly anyway. I will remove it.